src/code_7bb60_len_41b0.c

```diff
@@ -69,37 +133,174 @@ f32 func_800E34D8(void) {
+
+    if (temp_64 != 0.0f) {
+        f32 x = playerStatus->position.x;
+        f32 y = playerStatus->position.y;
+        f32 z = playerStatus->position.z;
+
+        do_lateral_collision(0, playerStatus, &x, &y, &z, temp_64, playerStatus->unk_88);
+
+        temp_64 -= (playerStatus->runSpeed / 10.0f);
+        playerStatus->position.x = x;
+        playerStatus->position.y = y;
+        playerStatus->position.z = z;
+
+        if (temp_64 < 0.0f) {
+            temp_64 = 0.0f;
+        }
+
+        (*(&playerStatus))->unk_64 = temp_64;
+    }
+}
 
 INCLUDE_ASM(s32, "code_7bb60_len_41b0", func_800E45E0);
 
 INCLUDE_ASM(s32, "code_7bb60_len_41b0", func_800E46C8);
 
 INCLUDE_ASM(s32, "code_7bb60_len_41b0", func_800E4744);
 
-INCLUDE_ASM(s32, "code_7bb60_len_41b0", func_800E4AD8);
+void func_800E4AD8(s32 arg0) {
+    Camera* currentCamera = &gCameras[gCurrentCameraID];
+
+    collision_check_player_intersecting_world(arg0, 0, gPlayerStatus.spriteFacingAngle - 90.0f + currentCamera->currentYaw);
+}
+
+void func_800E4B40(s32 arg0, f32* arg1, f32* arg2, f32* arg3) {
+    Camera* currentCamera = &gCameras[gCurrentCameraID];
 
-INCLUDE_ASM(s32, "code_7bb60_len_41b0", func_800E4B40);
+    func_800E4404(arg0, 0, gPlayerStatus.spriteFacingAngle - 90.0f + currentCamera->currentYaw, arg1, arg2, arg3);
+}
 
 INCLUDE_ASM(s32, "code_7bb60_len_41b0", func_800E4BB8);
 
-INCLUDE_ASM(s32, "code_7bb60_len_41b0", func_800E4F10);
+void func_800E4F10(void) {
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    s32 tempB = 0;
+    f32 yaw = playerStatus->targetYaw;
+    f32 x = playerStatus->position.x;
+    f32 y = playerStatus->position.y;
+    f32 z = playerStatus->position.z;
+    s32 temp = func_800DF15C(&gPlayerStatus, &x, &y, &z, 0, yaw, &tempB);
+
+    playerStatus->position.x = x;
+    playerStatus->position.z = z;
 
-INCLUDE_ASM(s32, "code_7bb60_len_41b0", check_input_midair_jump);
+    if ((tempB != 0) && (temp < 0) && (playerStatus->actionState != 0x18) && (playerStatus->currentSpeed != 0.0f)) {
+        set_action_state(0x18);
+    }
+}
+
+void check_input_midair_jump(void) {
+    if (!(gPlayerStatus.flags & 0x800018) &&
+        !(gPlayerStatus.animFlags & 0x4001) &&
+        gPlayerStatus.unk_C2 >= 6 &&
+        gPlayerStatus.decorationList < 0x12 &&
+        gPlayerStatus.pressedButtons & A_BUTTON) {
+
+        switch (gPlayerData.bootsLevel) {
+            case 0:
+                break;
+            case 1:
+                set_action_state(ActionState_SPIN_JUMP);
+                gPlayerStatus.flags |= 8;
+                break;
+            case 2:
+                set_action_state(ActionState_ULTRA_JUMP);
+                gPlayerStatus.flags |= 8;
+                break;
+        }
+    }
+}
 
 PartnerID get_current_partner_id(void) {
     return gPlayerData.currentPartner;
 }
 
-INCLUDE_ASM(s32, "code_7bb60_len_41b0", func_800E5098);
+void func_800E5098(s32 arg0) {
+    if (((*gGameStatusPtr)->frameCounter % arg0) == 0) {
```

Comment:
extra parens

Reply:
I disagree here. Even if one knows the order of operations, I don't like having `long_expression % y == z`